src/zenml/cli/service_accounts.py

```diff
@@ -0,0 +1,515 @@
+from zenml.cli.utils import list_options
+from zenml.client import Client
+from zenml.console import console
+from zenml.enums import CliCategories, StoreType
+from zenml.exceptions import EntityExistsError
+from zenml.logger import get_logger
+from zenml.models import (
+    APIKeyFilterModel,
+    ServiceAccountFilterModel,
+)
+
+logger = get_logger(__name__)
+
+
+def _create_api_key(
+    service_account_name_or_id: str,
+    name: str,
+    description: Optional[str],
+    set_key: bool = False,
+) -> None:
+    """Create an API key.
+
+    Args:
+        service_account_name_or_id: The name or ID of the service account to
+            which the API key should be added.
+        name: Name of the API key
+        description: The API key description.
+        set_key: Configure the local client with the generated key.
+    """
+    client = Client()
+    zen_store = client.zen_store
+    if not zen_store.TYPE == StoreType.REST:
+        raise NotImplementedError(
+            "API key configuration is only supported if connected to a "
+            "ZenML server."
+        )
+
+    with console.status(f"Creating API key '{name}'...\n"):
+        try:
+            api_key = client.create_api_key(
+                service_account_name_id_or_prefix=service_account_name_or_id,
+                name=name,
+                description=description or "",
+                set_key=set_key,
+            )
+        except (KeyError, EntityExistsError) as e:
+            cli_utils.error(str(e))
+
+        cli_utils.declare(f"Successfully created API key `{name}`.")
+
+    if set_key and api_key.key:
+        client.set_api_key(api_key.key)
+        cli_utils.declare(
+            "The local client has been configured with the new API key."
+        )
+    else:
+        cli_utils.declare(
+            f"The API key value is: '{api_key.key}'\nPlease store it safely as "
+            "it will not be shown again.\nTo configure a ZenML client to use "
+            "this API key, run:\n\n"
+            f"zenml connect --url {zen_store.config.url} --api-key \\\n"
+            f"    '{api_key.key}'\n"
+        )
+
+
+@cli.group(cls=TagGroup, tag=CliCategories.IDENTITY_AND_SECURITY)
+def service_account() -> None:
+    """Commands for service account management."""
+
+
+@service_account.command(
+    "create", help="Create a new service account and optional API key."
+)
+@click.argument("service_account_name", type=str, required=True)
+@click.option(
+    "--skip-api-key",
+    help=("Skip creating an API key for the service account."),
+    is_flag=True,
+)
+@click.option(
+    "--set-key",
+    help=("Configure the local client to use the generated API key."),
+    is_flag=True,
+)
+@click.option(
+    "--role",
+    "-r",
+    "initial_role",
+    help="Give the service account an initial role.",
+    required=False,
+    type=str,
+    default="admin",
+)
+def create_service_account(
+    service_account_name: str,
+    skip_api_key: bool = False,
```

Comment:
create_api_key instead of skip, seems more natural to me imo. Less mental gymnastics required to understand whats going on ( don’t skip being a sort of double negative )